File: BackEnd/SpringBootServer/src/main/java/com/ssafy/backend/domain/news/entity/NewsKeywordHistory.java
package com.ssafy.backend.domain.news.entity;

import com.ssafy.backend.domain.member.entity.Member;
import com.ssafy.backend.global.common.entity.BaseEntity;
import jakarta.persistence.*;
import lombok.*;

@Getter
@NoArgsConstructor(access = AccessLevel.PROTECTED)
@AllArgsConstructor(access = AccessLevel.PROTECTED)
@Builder
@Entity
public class NewsKeywordHistory extends BaseEntity {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @Column(columnDefinition = "INT UNSIGNED")
    private Long id;

    @Column(nullable = false)
    private Integer readCount = 0;

    @ManyToOne
    @JoinColumn(name = "member_id")
    private Member member;

    @OneToOne
    @JoinColumn(name = "keyword_id")
    private Keyword keyword;
}

File: BackEnd/FastApiServer/models/news_keyword_history.py
from sqlalchemy import Column, Integer, Float, ForeignKey
from sqlalchemy.orm import relationship
from database import Base

class NewsKeywordHistory(Base):
    __tablename__ = "news_keyword_history"
    id = Column(Integer, primary_key=True, index=True)
    read_count = Column(Integer, default=0, nullable=False)
    member_id = Column(Integer, ForeignKey("member.id"))
    member = relationship("Member", back_populates="news_keyword_histories")
    keyword_id = Column(Integer, ForeignKey("keyword.id"))
    keyword = relationship("Keyword", back_populates="news_keyword_histories")